accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,229 @@
+  - auth_type: `cloudrmt`
+
+We also need to save the repository base URL in `suseCredentials` table. URL from table `susesccrepository` is pointing to SCC and we need to connect to the RMT cloud servers. For that we need to save URL prefix, including hostname.
+Example of RMT base URL:
+- URL on pay-as-you-go instance: `https://host/repo/SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+- Sub-string used to search repository on `susSccrespository` table: `/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+- Repository base URL to save on `suseCredentials` table: `https://host/repo`
+
+
+### Implementation steps - Add product with setup wizard
+
+When a product is added in the setup wizard table `rhncontentsource` is populated according to the authentication mechanism defined in table `susesccrepositoryauth` using the repository url defined in table `susesccrepository`.
+
+We need to adapt this feature to consider the new `cloudrmt` authentication mechanism. URL that will be inserted in `rhncontentsource` will have the following rules:
+  - Concatenation of base URL from `suseCredentials` with the url path from `susesccrepository`
+  - Authentication will use the same mechanism as basic authentication, where id of `suseCredentials` is passed as query string in the repository URL
+
+
+### Implementation steps - reposync
+
+Reposync is already able to deal with the basic authentication mechanism. It receives the id of a `suseCredentials` record and loads the basic authentication data.
+This mechanism will be enhanced to consider also the new column with the authentication headers. The following changes need to be implemented:
+  - Modify method [`_url_with_repo_credentials`](https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/reposync.py#L1785) to also load and return the http authentication headers from table `suseCredentials`
+  - Enhance [repository plugins](https://github.com/uyuni-project/uyuni/tree/master/backend/satellite_tools/repo_plugins) to receive http headers
+  - Enhance reposync [zypper plugin](https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/spacewalk-extra-http-headers) to receive http header. We have two implementation options:
+    - Add http headers to zypper `.repo` file as extra field. Verify if the implementation is possible and doesn't impact zypper command.
+    - Write the headers to a temporary file in the same location of zypper `.repo` file
+
+Another mechanism for [authentication headers](https://github.com/uyuni-project/uyuni/pull/2136) is already in place which loads headers from a configuration file. We should analyze if it can be merged with the new implementation.
+
+
+### Implementation steps - Refresh repository authentication data
+
+RMT authentication data have a time to live (TTL). For this reason, we need to load new authentication data from the pay-as-you-go instance periodically. For that, a new taskomatic job needs to be defined to:
```

Comment:
Why a `new` job? Why not doing it together with the nightly refresh job. This would guarantee that we check one time per day and exactly before we refresh the data from SCC. That job also start the reposync of all vendor channels which would guarantee that the data are up to date.

Reply:
Great hint Micheal. I will have a look on it, and see how that would be implemented and then upate this RFC according.

Reply:
We moved to the solution of a new job to retrieve the ssh data from the payg instance periodically. With this approach we expect that even a call to reposybc on the command line will also work.